src/modules/cmdpal/Microsoft.CmdPal.UI/ExtViews/ListPage.xaml.cs

```diff
@@ -43,6 +45,16 @@ public ViewModelLoadedState LoadedState
     public static readonly DependencyProperty LoadedStateProperty =
         DependencyProperty.Register(nameof(LoadedState), typeof(ViewModelLoadedState), typeof(ListPage), new PropertyMetadata(ViewModelLoadedState.Loading));
 
+    public string ErrorMessage
+    {
+        get => (string)GetValue(ErrorMessageProperty);
+        set => SetValue(ErrorMessageProperty, value);
+    }
+
+    // Using a DependencyProperty as the backing store for LoadedState.  This enables animation, styling, binding, etc...
+    public static readonly DependencyProperty ErrorMessageProperty =
+        DependencyProperty.Register(nameof(ErrorMessage), typeof(string), typeof(ListPage), new PropertyMetadata(string.Empty));
```

Comment:
Just a question, we don't want this just as a property on the `ListViewModel` itself do we? As we set call the message from the `ListViewModel` itself already... We could put the loading state there too and then just do binding/converter/UI View magic to have all the things display as we expect vs. having some of this logic in our view.

Generally, we want to use messages when we're communicating across boundaries (View <-> ViewModel) or different components that have different responsibilities.

Just want to ensure we're not introducing more complexity than we need here or following some early prototyping setup that we did that we want to now pivot/rethink. But this still works fine if we want to separate out concerns, just here to provide us options on how we think about these scenarios. 🙂

Reply:
That's what I originally thought, and then got myself confused seeing `LoadedState` as a dependency property 😅

Reply:
Putting it here so it's somewhere before the burrito screams more

IShowExceptionContext. Pass that into the ExtensionObject ctor. Every VM that's a VM hosted by a page then calls ShowException on the context, so that it goes to the page that owns it